### ISSUE TITLE
model path

### ISSUE THREAD

**Issue by 123dddwn** [zty557/TADSR#8]
parser.add_argument("--pretrained_model_name_or_path", default=None, type=str)
parser.add_argument("--pretrained_model_name_or_path_reg", default=None, type=str)
parser.add_argument("--pretrained_model_name_or_path_fix", default=None, type=str)
parser.add_argument("--pretrained_lora_path", default=None, type=str)
您好，第一个路径我知道填SD2.1的预训练权重参数（vae替换为time_vae），后面三个路径不太清楚是什么。

**Comment by zty557** [comment]
这些是之前测试时遗留的参数，pretrained_model_name_or_path_reg填写和pretrained_model_name_or_path一样的路径即可，剩下两个应该可以忽略

**Comment by 123dddwn** [comment]
class TADSR_gen(torch.nn.Module):
    def __init__(self, args):
        super().__init__()

        self.tokenizer = AutoTokenizer.from_pretrained(args.pretrained_model_name_or_path, subfolder="tokenizer")
        self.text_encoder = CLIPTextModel.from_pretrained(args.pretrained_model_name_or_path, subfolder="text_encoder").cuda()
        self.noise_scheduler = DDPMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
        self.noise_scheduler.set_timesteps(1, device="cuda")
        self.noise_scheduler.alphas_cumprod = self.noise_scheduler.alphas_cumprod.cuda()
        self.args = args

        self.vae, self.lora_vae_modules_encoder = initialize_vae_time(self.args)
        self.unet, self.lora_unet_modules_encoder, self.lora_unet_modules_decoder, self.lora_unet_others = initialize_unet(self.args)
        
        self.load_pretrained_model()
        
        self.lora_rank_unet = self.args.lora_rank
        self.lora_rank_vae = self.args.lora_rank

        self.unet.to("cuda")
        self.vae.to("cuda")
        self.timesteps = torch.tensor([1], device="cuda").long()
        self.text_encoder.requires_grad_(False)

    def load_pretrained_model(self):
        """加载预训练的LoRA权重
        
        Args:
            pretrained_model_name_or_path (str): 预训练模型的路径
        """
        if hasattr(self.args, 'pretrained_lora_path') and self.args.pretrained_lora_path is not None:
            print(f"Loading pretrained LoRA weights from {self.args.pretrained_lora_path}")
            
            # 加载预训练权重
            pretrained_weights = torch.load(self.args.pretrained_lora_path, map_location="cuda")
            
            # 加载UNet的LoRA权重
            if "state_dict_unet" in pretrained_weights:
                print("Loading UNet LoRA weights...")
                for n, p in self.unet.named_parameters():
                    if "lora" in n and n in pretrained_weights["state_dict_unet"]:
                        p.data.copy_(pretrained_weights["state_dict_unet"][n])
                        print(f"Loaded UNet weight: {n}")
                
                # 加载conv_in权重
                if "conv_in.weight" in pretrained_weights["state_dict_unet"]:
                    self.unet.conv_in.weight.data.copy_(pretrained_weights["state_dict_unet"]["conv_in.weight"])
                    print("Loaded UNet conv_in.weight")
                if "conv_in.bias" in pretrained_weights["state_dict_unet"]:
                    self.unet.conv_in.bias.data.copy_(pretrained_weights["state_dict_unet"]["conv_in.bias"])
                    print("Loaded UNet conv_in.bias")
            
            # 加载VAE的LoRA权重
            if "state_dict_vae" in pretrained_weights:
                print("Loading VAE LoRA weights...")
                for n, p in self.vae.named_parameters():
                    if "lora" in n and n in pretrained_weights["state_dict_vae"]:
                        p.data.copy_(pretrained_weights["state_dict_vae"][n])
                        print(f"Loaded VAE weight: {n}")
            
            # 加载VAE的time权重（如果存在）
            if "state_dict_vae_time" in pretrained_weights:
                print("Loading VAE time weights...")
                for n, p in self.vae.named_parameters():
                    if "time" in n and n in pretrained_weights["state_dict_vae_time"]:
                        p.data.copy_(pretrained_weights["state_dict_vae_time"][n])
                        print(f"Loaded VAE time weight: {n}")
            
            print("Successfully loaded pretrained LoRA weights")
        else:
            print("No pretrained LoRA path specified, using random initialization")
但是这里如果不传入Lora的权重参数，就会随机初始化，而随机初始化会不会在相同的iters下影响指标结果呢？

**Comment by zty557** [comment]
这个是有可能的，但是影响应该不是很大，一般挂lora训练权重都是随机初始化的，这个功能之前应该是resume的时候会用
